src/api/app/models/token/workflow.rb

```diff
@@ -25,14 +25,17 @@ def call(options)
     raise Token::Errors::MissingPayload, 'A payload is required' if @scm_webhook.payload.blank?
 
     workflow_run.update(response_url: @scm_webhook.payload[:api_endpoint])
-    yaml_file = Workflows::YAMLDownloader.new(@scm_webhook.payload, token: self).call
-    @workflows = Workflows::YAMLToWorkflowsService.new(yaml_file: yaml_file, scm_webhook: @scm_webhook, token: self, workflow_run: workflow_run).call
+    # We can't get the yaml from a ping event, because the payload doesn't contain all the necessary information
+    unless ping?
+      yaml_file = Workflows::YAMLDownloader.new(@scm_webhook.payload, token: self).call
+      @workflows = Workflows::YAMLToWorkflowsService.new(yaml_file: yaml_file, scm_webhook: @scm_webhook, token: self, workflow_run: workflow_run).call
+    end
 
     return validation_errors unless validation_errors.none?
```

Comment:
`validation_errors` is checking for validation errors in workflows, so this part of the validation should be skipped.